5-AccessControl/2-call-api-groups/API/auth/overage.js

```diff
@@ -57,70 +58,40 @@ const callGraph = async (oboToken, endpoint) => {
         const response = await axios.default.get(endpoint, options);
         return response.data;
     } catch (error) {
-        console.log(error)
+        console.log(error);
         return error;
     }
-}
+};
 
 const handlePagination = async (oboToken, nextPage, userGroups) => {
-
     try {
         const graphResponse = await callGraph(oboToken, nextPage);
 
         graphResponse.value.map((v) => userGroups.push(v.id));
 
         if (graphResponse['@odata.nextLink']) {
-            return await handlePagination(oboToken, graphResponse['@odata.nextLink'], userGroups)
+            return await handlePagination(oboToken, graphResponse['@odata.nextLink'], userGroups);
         } else {
-            return userGroups
+            return userGroups;
         }
     } catch (error) {
         console.log(error);
     }
-
-}
+};
 
 const checkAccess = (req, res, next) => {
     const accessMatrix = config.accessMatrix;
     const groups = res.locals.groups;
-
-    if (req.path.includes(accessMatrix.todolist.path)) {
-        if (accessMatrix.todolist.methods.includes(req.method)) {
-
-            let intersection = accessMatrix.todolist.groups
-                .filter(group => groups.includes(group));
-
-            if (intersection.length < 1) {
-                return res.status(403).json({ error: 'User does not have the group' });
-            } else {
-                return next();
-            }
-        } else {
-            return res.status(403).json({ error: 'Method not allowed' });
-        }
-    } else if (req.path.includes(accessMatrix.dashboard.path)) {
-        if (accessMatrix.dashboard.methods.includes(req.method)) {
-
-            let intersection = accessMatrix.dashboard.groups
-                .filter(group => groups.includes(group));
-
-            if (intersection.length < 1) {
-                return res.status(403).json({ error: 'User does not have the group' });
-            } else {
-                return next();
-            }
-        } else {
-            return res.status(403).json({ error: 'Method not allowed' });
-        }
-    } else {
-        return res.status(403).json({ error: 'Unrecognized path' });
+    if (!requestHasRequiredAttributes(accessMatrix, req.path, req.method, groups)) {
+        return res.status(403).json({ error: 'User does not have the group, method or path' });
     }
-}
+    next();
+};
 
 const handleOverage = async (req, res, next) => {
     console.log('going through overage');
```

Comment:
```suggestion
    console.log('going through overage');
```

Can probably remove this